File: Question 0034 - Find First and Last Position of Element in Sorted Array/src/Solution.java
class Solution {
    public int[] searchRange(int[] nums, int target)
    {
        int start = 0;
        int end = nums.length-1;

        int mid = start + (end-start)/2 ;

        int first = -1, last = -1;
        while (start <= end)
        {
            if(nums[mid] == target)
            {
                first = mid;
                end = end-1;
            }
            else if(target > nums[mid]) start = mid+1;
            else end = mid-1;

            mid = start + (end-start)/2 ;
        }

        start = 0;
        end = nums.length-1;
        mid = start + (end-start)/2 ;

        while (start <= end)
        {
            if(nums[mid] == target)
            {
                last = mid;
                start = start+1;
            }
            else if(target > nums[mid]) start = mid+1;
            else end = mid-1;

            mid = start + (end-start)/2 ;
        }

        return new int[]{first,last};
    }
}

File: Question 0034 - Find First and Last Position of Element in Sorted Array/src/Solution.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution 
{
public:
    vector<int> searchRange(vector<int>& nums, int target) 
    {
        int left = -1, right = -1;
        int l = 0, r = nums.size()-1;

        while(l <= r)
        {
            int mid = l + (r-l)/2;

            if(nums[mid] >= target)
            {
                if(nums[mid] == target) left = mid;
                r = mid-1;
            }
            else l = mid+1;
        }

        l = 0, r = nums.size()-1;
        while(l <= r)
        {
            int mid = l + (r-l)/2;

            if(nums[mid] <= target)
            {
                if(nums[mid] == target) right = mid;
                l = mid+1;
            }
            else r = mid-1;
        }

        return {left, right};
    }
};